File: android/src/main/java/com/lightningkite/khrysalis/observables/binding/View.binding.actual.kt
package com.lightningkite.khrysalis.observables.binding

import android.view.View
import com.lightningkite.khrysalis.observables.*
import com.lightningkite.khrysalis.rx.removed
import com.lightningkite.khrysalis.rx.until

/**
 *
 * Binds the visibility of the view to the observable provided.
 * If the value if false, the view will not be visible, nor intractable, though the
 * view will still take up space.
 *
 */

fun View.bindVisible(observable: ObservableProperty<Boolean>) {
    observable.subscribeBy { value ->
        this.visibility = if (value) View.VISIBLE else View.INVISIBLE
    }.until(this.removed)
}


/**
 *
 * Binds the existance of the view to the observable provided.
 * If the value if false, the view will not exist, meaning it will not be visible,
 * it will not be intractable, and it will not take up any space.
 *
 */
fun View.bindExists(observable: ObservableProperty<Boolean>) {
    observable.subscribeBy { value ->
        this.visibility = if (value) View.VISIBLE else View.GONE
    }.until(this.removed)
}

File: ios/Khrysalis/src/observables/binding/View.binding.actual.swift
//Stub file made with Khrysalis 2 (by Lightning Kite)
import Foundation
import UIKit


//--- View.bindVisible(ObservableProperty<Boolean>)
public extension UIView {
    func bindVisible(_ observable: ObservableProperty<Bool>) -> Void {
        observable.subscribeBy { ( value) in
            self.isHidden = !value
        }.until(self.removed)
    }
    func bindVisible(observable: ObservableProperty<Bool>) -> Void {
        return bindVisible(observable)
    }
}

//--- View.bindExists(ObservableProperty<Boolean>)
public extension UIView {
    func bindExists(_ observable: ObservableProperty<Bool>) -> Void {
        observable.subscribeBy { ( value) in
            self.includeInLayout = value
            self.isHidden = !value
        }.until(self.removed)
    }
    func bindExists(observable: ObservableProperty<Bool>) -> Void {
        return bindExists(observable)
    }
}
